modules/data.atmosphere/R/met_temporal_downscale.Gaussian_ensemble.R

```diff
@@ -0,0 +1,398 @@
+  # Reading in the training data
+  train <- list()
+  tem <- ncdf4::nc_open(train_met)
+  dim <- tem$dim
+  for (j in seq_along(var$CF.name)) {
+    if (exists(as.character(var$CF.name[j]), tem$var) == FALSE) {
+      train[[j]] <- NA
+    } else {
+      train[[j]] <- ncdf4::ncvar_get(tem, as.character(var$CF.name[j]))
+    }
+  }
+  lat_train <- as.numeric(ncdf4::ncvar_get(tem, "latitude"))
+  lon_train <- as.numeric(ncdf4::ncvar_get(tem, "longitude"))
+  ncdf4::nc_close(tem)
+  
+  train <- data.frame(train)
+  colnames(train) <- var$CF.name
+  if (all(is.na(train$air_temperature_max))) {
+    train$air_temperature_max <- train$air_temperature
+  }
+  if (all(is.na(train$air_temperature_min))) {
+    train$air_temperature_min <- train$air_temperature
+  }
+  # Reading in the source data
+  source <- list()
+  tem <- ncdf4::nc_open(input_met)
+  for (j in seq_along(var$CF.name)) {
+    if (exists(as.character(var$CF.name[j]), tem$var) == FALSE) {
+      source[[j]] <- NA
+    } else {
+      source[[j]] <- ncdf4::ncvar_get(tem, as.character(var$CF.name[j]))
+    }
+  }
+  ncdf4::nc_close(tem)
+  
+  source <- data.frame(source)
+  colnames(source) <- var$CF.name
+  
+  # Default downscale will be to the resolution of the training dataset
+  reso <- 24/(nrow(train)/365)
+  reso_len <- nrow(train)
+  
+  # If a source doesn't have leap days, we need to remove them to ensure equal
+  # lengths
+  if (lubridate::leap_year(year) == TRUE) 
+  {
+    if (length(source$air_temperature)%%366 > 0) {
+      if (length(train$air_temperature)%%366 == 0) {
+        train <- train[1:365 * (nrow(train)/366)]
+      }
+      eph_year <- 2006
+    }
+  }  #chose a non-leap year to use for daylength calculations if we don't have the 
+  if (lubridate::leap_year(eph_year) == TRUE) {
+    sp <- 366
+  }
+  if (lubridate::leap_year(eph_year) == FALSE) {
+    sp <- 365
+  }
+  
+  # Now we start a for loop for the ensemble members and begin downscaling. A
+  # random normal distribution is used to downscale as so;
+  # (mean <- value of source data) (sd <- +/- window_days of train data at the
+  # same time intervals) 
+  for (e in seq_along(1:ensemble_members)) {
+    
+    div <- nrow(train)/nrow(source)  #tells us how many values need to be generated (via downscaling) from each source value
+    sd_step <- nrow(train)/sp  #allows us to step through each window at specific times
+    
+    # Temperature For temperature, we create a random normal distribution with the
+    # mean of the source value and the standard deviation of the training value
+    # within the window, and step through that window sequentially so that we are
+    # only using the standard devaition of values recorded at the same time as the
+    # value we are generating.
+    temp <- c("air_temperature", "air_temperature_max", "air_temperature_min")
+    df <- data.frame()
+    for (u in temp) {
+      woo <- vector()
+      a <- as.numeric(train[[u]])
+      sour <- as.numeric(source[[u]])
```

Comment:
you might want to think about more meaningful variable names -- I see that sour = source, but just 'a' for training data? No clue what woo is going to be.